From 14478dbf57ce8a8f06d299a7a4ad45fc05a60203 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
 <marmarek@invisiblethingslab.com>
Date: Tue, 21 Apr 2015 03:56:36 +0200
Subject: [PATCH 2/2] libxl: do not start qemu in dom0 if possible
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Organization: Invisible Things Lab
Cc: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>

After previous patch, the only trigger for dom0 qemu is number of
consoles. So if no additional serial console is defined, drop
save/restore stubdom channels as a price for no qemu in dom0.

Signed-off-by: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>
---
 tools/libxl/libxl_dm.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tools/libxl/libxl_dm.c b/tools/libxl/libxl_dm.c
index 52e7592..86f16bc 100644
--- a/tools/libxl/libxl_dm.c
+++ b/tools/libxl/libxl_dm.c
@@ -1083,6 +1083,9 @@ static void spawn_stub_launch_dm(libxl__egc *egc,
 
     if (guest_config->b_info.u.hvm.serial)
         num_console++;
+    else
+        /* Qubes hack */
+        num_console = 1;
 
     console = libxl__calloc(gc, num_console, sizeof(libxl__device_console));
     if (!console) {
-- 
2.1.0

